apps/emqx_bridge_kafka/src/emqx_bridge_kafka.erl

```diff
@@ -459,3 +460,11 @@ consumer_topic_mapping_validator(TopicMapping = [_ | _]) ->
         false ->
             {error, "Kafka topics must not be repeated in a bridge"}
     end.
+
+producer_strategy_key_validator(#{
+    <<"partition_strategy">> := key_dispatch,
+    <<"message">> := #{<<"key">> := ""}
```

Comment:
question: is it possible to configure empty string from dashboard ?

Reply:
I've just tested it with dashboard `e1.0.7` (without this patch) and it actually doesn't allow empty strings for the key.  When I tried to set the field to empty, it ends up setting it to the default value (`${.clientid}`).  :thinking: